File: csharp/perfect-numbers/PerfectNumbers.cs
using System;

public enum Classification
{
    Perfect,
    Abundant,
    Deficient
}

public static class PerfectNumbers
{
    public const int Mike = 42;
    public static Classification Classify(int number)
    {
        throw new NotImplementedException();
    }
}


namespace Bob
{
    public static class PerfectNumbers
    {
        public const int John = global::PerfectNumbers.Mike;
        public static void Foo()
        {
        
        }
    }
}

File: haskell/perfect-numbers/src/PerfectNumbers.hs
module PerfectNumbers (classify, Classification(..)) where

data Classification = Deficient | Perfect | Abundant deriving (Eq, Show)

classify :: Int -> Maybe Classification
classify n
    | n <= 0 = Nothing
    | otherwise = if result == n then Just Perfect
         else if result > n then Just Abundant
         else Just Deficient
        where result = sum [x | x <- [1..n`div`2], n `mod` x == 0]


-- classify :: Int -> Maybe Classification
-- classify n = if (floor (sqrt (fromIntegral n ))) > 10 then Nothing else Nothing